core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -484,14 +484,15 @@ private[kafka] class Processor(val id: Int,
   protected[network] def sendResponse(response: RequestChannel.Response, responseSend: Send) {
     trace(s"Socket server received response to send, registering for write and sending data: $response")
     val channel = selector.channel(responseSend.destination)
```

Comment:
We use `responseSend.destination` here and `response.request.connectionId` in the other two cases. They have the same value, but it would be clearer if we use the same.